File: src/HeadphoneStore.BackOffice/src/apis/admin/category.api.ts
import { URL } from ".."

export const BASE_URL = `${URL}/Category`

export const CATEGORY_API = {
  CREATE: `${BASE_URL}/create`,
  UPDATE: `${BASE_URL}`,
  DELETE: `${BASE_URL}`,
  PAGINATION: `${BASE_URL}/pagination`,
  ALL_SUB: `${BASE_URL}/all-sub`,
  GET_ALL_FIRST_LEVEL: `${BASE_URL}/all-first-level`,
}

File: src/HeadphoneStore.StoreFrontEnd/Apis/Endpoints/CategoryApi.cs
namespace HeadphoneStore.StoreFrontEnd.Apis.Endpoints;

public class CategoryApi
{
    /// <summary>
    /// [AllowAnonymous]
    /// <br />
    /// Description: Get all categories with sub-categories.
    /// <br />
    /// Sample endpoint: https://localhost:8081/api/v1/Category/all
    /// </summary>
    public const string GetAllCategories = "Category/all-first-level";
    
    /// <summary>
    /// [AllowAnonymous]
    /// <br />
    /// Description: Get all categories with sub-categories.
    /// <br />
    /// Sample endpoint: https://localhost:8081/api/v1/Category/all-with-sub
    /// </summary>
    public const string GetAllCategoriesWithSub = "Category/all-with-sub";
}
